packages/SwingSet/docs/networking.md

```diff
@@ -56,53 +57,49 @@ You can ask the `Port` object this returns for its local address, which is espec
+Once you have a `Connection` object, you send data by calling its `send()` method:
 
 ```js
 connection.send('data');
 ```
 
-`send` actually returns a Promise (for more `Bytes`), which contains the ACK data for this message.  NOTE: The type of this data and ACK is currently a string.  Ideally we would also accept Node.js `Buffer` objects, or Javascript `ArrayBuffer` and `TypedArray` objects, but unfortunately neither can be serialized by our current inter-vat marshalling code.
+`send()` actually returns a Promise (for more `Bytes`), which contains the ACK data for this message.  NOTE: The type of this data and ACK is currently a string.  Ideally we would also accept Node.js `Buffer` objects, or Javascript `ArrayBuffer` and `TypedArray` objects, but unfortunately neither can be serialized by our current inter-vat marshalling code.
 
-## Receiving Data: The ConnectionHandler
+## Receiving Data
 
 You must provide each open connection with a `ConnectionHandler` object, where you write methods that will be called when various things happen to the connection. You can share a single handler object between multiple connections, if you like, or you can make a separate one for each.
 
 You can omit any of the methods and those events will simply be ignored. All these methods include the Connection object as the first argument, and the `ConnectionHandler` itself as the last argument, which might help if you want to share a common handler function among multiple connections.
 
 * `onOpen(connection, handler)`: this is called when the connection is established, which tells you that the remote end has successfully accepted the connection request
-* `onReceive(connection, packetBytes, handler)`: this is called each time the remote end sends a packet full of data
-* `onClose(connection, reason, handler)`: this is called when the connection is closed, either because one side wanted it to close, or because an error occurred
-
-The `reason` in `onclose` is optional, as in it may be `undefined`.
+* `onReceive(connection, packetBytes, handler)`: this is called each time the remote end sends a packet of data
+* `onClose(connection, reason, handler)`: this is called when the connection is closed, either because one side wanted it to close, or because an error occurred. `reason` may be `undefined`.
 
-`onReceive` is the most important method. Each time the remote end sends a packet, your `onReceive` method will be called with the data inside that packet (currently as a String, but ideally as an ArrayBuffer with a custom `.toString()` method with an optional `encoding` argument (default `'latin1'`), so that it can contain arbitrary bytes).
+`onReceive()` is the most important method. Each time the remote end sends a packet, your `onReceive()` method will be called with the data inside that packet (currently as a String, but ideally as an ArrayBuffer with a custom `toString(encoding='latin1')` method so that it can contain arbitrary bytes).
```

Comment:
I'm curious, why is this "latin1" rather than "utf8"?

Reply:
> why is this "latin1" rather than "utf8"?

All combinations of octet values from `\x00` to `\xff` are accommodated in `latin1` encoding and round trip in the low part of a UTF16 code point.  Some of those combinations are invalid `utf8` and IIUC don't round-trip in UTF16 correctly.  I learned this long ago enough that it may have just been a quirk of the system I was working with.  I'd be happy to be reeducated if you know better.

Reply:
That is true, but losslessly converting an ArrayBuffer of arbitrary octets into a string of [U+0000, U+00FF] code points doesn't seem particularly useful (binary treatment wouldn't need such conversion, and a sequence of _non_-arbitrary octets could be use a more relevant conversion). Regardless, I'm happy to leave this for now.